code.js

```diff
@@ -1,3 +1,38 @@
+/*
+Author: Isaiah Elston
+Project: Binary Search
+
+NOTE: I have very little experience with JavaScript, so I am teaching myself as I go. Therefore, I will be including my sources
+      to seemingly trivial tasks in the beginning of the class...
+
+SOURCES:
+
+To learn more about the Binary Search function (I did not take COSC 2030 at UW and have not run into Binary Search prior to 
+this course...)
+https://www.geeksforgeeks.org/binary-search/#
+
+For Math.floor()
+https://stackoverflow.com/questions/4228356/how-to-perform-an-integer-division-and-separately-get-the-remainder-in-javascr
+*/
+
 function binarySearch(list, element) {
-    return -1;
-}
+    let leftBound = 0;
+    let rightBound = list.length;
+
+    do {
+        let middleIndex = Math.floor(leftBound + ((rightBound - leftBound) / 2));
+        if (list[middleIndex] == element && 
+            list[middleIndex] !== list[leftBound]) {
+            return middleIndex;
+        } else if (list[middleIndex] == element &&
+                   list[middleIndex] == list[leftBound]) {
+                    return leftBound;
+        }else if (list[middleIndex] > element) {
+            rightBound = middleIndex--;
+        } else if (list[middleIndex] < element) {
+            leftBound = middleIndex++;
+        } else {
+            return -1;
+        }
+    } while (rightBound >= 1);
```

Comment:
This doesn't look quite right. What happens if the right bound is less than the left?